web_src/css/modules/animations.css

```diff
@@ -1,36 +1,31 @@
-@keyframes isloadingspin {
-  0% { transform: translate(-50%, -50%) rotate(0deg); }
-  100% { transform: translate(-50%, -50%) rotate(360deg); }
-}
-
 .is-loading {
   pointer-events: none !important;
-  position: relative !important;
   overflow: hidden !important;
+
+  background-image: url("../../svg/spinner-ring.svg");
```

Comment:
No URL embeds, those would incur extra HTTP requests we don't need/want. You can insert the whole SVG into a custom property like done here:

https://github.com/go-gitea/gitea/blob/c2b6897e35ccdebc4b49cfc7fb165ee654c8e55d/web_src/css/base.css#L15C7-L15C7

Likely, you will need to apply it via `mask-image` because of https://stackoverflow.com/questions/49381592 like done here:

https://github.com/go-gitea/gitea/blob/c2b6897e35ccdebc4b49cfc7fb165ee654c8e55d/web_src/css/modules/select.css#L19-L22

Reply:
Working on it! One question about this though, how do I properly credit the author? I used the animations from the repos you linked, and slightly edited them myself for the static versions, so I need to keep the original author's copyright mention… Where do I put that? At the top of the css file itself?

Reply:
Good point, we have no good place to credit, but you can add a link here (rename "UI Frameworks" to just "UI":

https://github.com/go-gitea/gitea/blob/7e382a55558d4a8719e8493de5455883a2003759/docs/content/index.en-us.md?plain=1#L286